brain/compartmentReportView.cpp

```diff
@@ -120,9 +56,22 @@ const CompartmentReportMapping& CompartmentReportView::getMapping() const
 }
 
 std::future<CompartmentReportFrame> CompartmentReportView::load(
-    const float timestamp)
+    double timestamp)
 {
+    const double start = _impl->report->getStartTime();
+    const double end = _impl->report->getEndTime();
+
+    if (timestamp < start || timestamp >= end)
+        throw std::logic_error("Invalid timestamp");
+
+    const double timestep = _impl->report->getTimestep();
+
+    size_t frameIndex = std::llround((timestamp - start) / timestep);
```

Comment:
The tests failures that I get are real bugs. The problem is that brion::CR uses single precision floats.
floor doesn't work properly with single precision floats:
       float t = 9.4, timestep = 0.1;
       std::cout << "round " << (size_t)std::round(t / timestep) << std::endl;
       std::cout << "floor " << (size_t)std::floor(t / timestep) << std::endl;
>> round 94
>> floor 93
        double t = 9.4, timestep = 0.1;
        std::cout << "round " << (size_t)std::round(t / timestep) << std::endl;
        std::cout << "floor " << (size_t)std::floor(t / timestep) << std::endl;
>> round 94
>> floor 94